syntax = "proto3";

package kalman.services;

import "google/protobuf/empty.proto";
import "kalman/data/data.proto";
import "kalman/data/motion_model.proto";

service KalmanFilterRPC {
  rpc Create(CreateRequest) returns (google.protobuf.Empty);
  rpc Tick(TickRequest) returns (TickResponse);
  rpc Get(GetRequest) returns (TickResponse);
}

message CreateRequest {
  string identifier = 1;
  kalman.motion_model.MotionModel motion_model = 2;
  // variance in distance measurement.
  float measurement_variance = 3;
}

message TickRequest {
  string identifier = 1;
  kalman.data.Measurement measurement = 2;
}

message TickResponse {
  string identifier = 1;
  kalman.data.Estimate estimate = 2;
}

message GetRequest { string identifier = 1; }
